pkg/logql/downstream.go

```diff
@@ -359,6 +361,42 @@ func (ev *DownstreamEvaluator) StepEvaluator(
 
 		return ConcatEvaluator(xs)
 
+	case *TopkMergeSampleExpr:
+		cur := e
+		var queries []DownstreamQuery
+		for cur != nil {
+			qry := DownstreamQuery{
+				Expr:   cur.DownstreamTopkSampleExpr.TopkSampleExpr,
+				Params: params,
+			}
+			if shard := cur.DownstreamTopkSampleExpr.shard; shard != nil {
+				qry.Shards = Shards{*shard}
+			}
+			queries = append(queries, qry)
+			cur = cur.next
+		}
+
+		results, err := ev.Downstream(ctx, queries)
```

Comment:
we should include "make downstream responses parsable before all reponses are returned" an item in our engine improvements list, since topk sketches should be mergable in any order we don't need to wait for all results here, we could have `DownstreamAsync` that takes a channel that results can be sent to?

Reply:
Won't do in this PR.